test/tangent_types/tangent.jl

```diff
@@ -78,6 +78,13 @@ end
         @test getindex(Tangent{Tuple{Float64}}(@thunk 2.0^2), 1) == 4.0
         @test getproperty(Tangent{Tuple{Float64}}(2.0), 1) == 2.0
         @test getproperty(Tangent{Tuple{Float64}}(@thunk 2.0^2), 1) == 4.0
+        
+        tang3 = Tangent{Tuple{Float64, String, Vector{Float64}}}(1.0, NoTangent(), @thunk [3.0] .+ 4)
```

Comment:
Should we test this behaviour explicitly?
```julia
julia> tang1 = Tangent{Tuple{Float64}}(1.0)
Tangent{Tuple{Float64}}(1.0,)

julia> Base.tail(tang1)
Tangent{Tuple{}}()
```

Reply:
Maybe this should just return NoTangent()?